### ISSUE TITLE
More efficient way to handle post objects in vuex

### ISSUE THREAD

**Issue by jrobson153** [funkhaus/vuepress#64]
Currently, we dump the entire contents of Rest Easy into the vuex store and then work off of that. In certain situations, such as the one outlined in issue https://github.com/funkhaus/Rest-Easy/issues/16 the vast majority of that data could potentially be redundant. 

In these situations the best solution is to find a cleaner way to output all that data at the server level. If vuepress does end up getting a large rest-easy payload however, we should put extra logic in place to reduce redundancy. The stress put on the front end can be considerable, because vue is attempting to be reactive to every single object even though most of them will never change.

Basically, it's important to keep the total number of serialized posts we store in vuex low.
